### PR TITLE
[TACHYON-1353] Wait for tachyonConf ports to be reset before tests

### PR DESCRIPTION
**Pull request by aaudiber**
https://tachyon.atlassian.net/browse/TACHYON-1353

### CODE REVIEW

**Comment by gpang** [comment]
This works because the LocalTachyonCluster will set the ports accordingly in TachyonConf after it starts the appropriate threads?

**Comment by aaudiber** [comment]
@gpang Yep, otherwise there's a race condition where the server is up so the tests start, but the `TachyonConf` hasn't been updated yet so the tests fail by trying to access the server on port 0.

**Comment by gpang** [comment]
Thanks for the fix!

**Comment by gpang** [comment]
LGTM

File: minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java [hunk 1]
```diff
@@ -108,14 +108,16 @@ private void waitForMasterReady() {
     long startTime = System.currentTimeMillis();
     String actionMessage = "waiting for master to serve web";
     LOG.info(actionMessage + ELLIPSIS);
+    // The port should be set properly after the server has started
     while (!NetworkAddressUtils.isServing(getMaster().getWebBindHost(),
-        getMaster().getWebLocalPort())) {
+        getMaster().getWebLocalPort()) || mMasterConf.getInt(Constants.MASTER_WEB_PORT) == 0) {
```
Comment (yupeng9) [inline minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java]: Should the second condition be merged into the `isServing` checking?

Comment (aaudiber) [inline minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java]: `isServing` is currently purely a network address util - it shouldn't need to worry about the way we reset `TachyonConf`s, which is more an implementation detail of how we start up the test cluster

Comment (yupeng9) [inline minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java]: ah, I see. Makes sense/

**Comment by yupeng9** [comment]
LGTM

**Pull request merged**